src/zitadel/mod.rs

```diff
@@ -0,0 +1,182 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use testcontainers::{
+    core::{wait::HttpWaitStrategy, ContainerPort, WaitFor},
+    Image,
+};
+
+const NAME: &str = "ghcr.io/zitadel/zitadel";
+const TAG: &str = "v2.69.10";
+const DEFAULT_MASTER_KEY: &str = "MasterkeyNeedsToHave32Characters";
+
+/// Port that the [`Zitadel`] container has internally
+/// Can be rebound externally via [`testcontainers::core::ImageExt::with_mapped_port`]
+///
+/// [`Zitadel`]: https://zitadel.com/
+pub const ZITADEL_PORT: ContainerPort = ContainerPort::Tcp(8080);
+
+// Zitadel testcontainer module
+//
+// This module provides a [Zitadel] container that can be used for testing purposes.
+// It uses the official Zitadel Docker image and configures it with a default master key
+// and in-memory database for quick testing.
+//
+// # Example
+// ```rust,no_run
+// use testcontainers_modules::zitadel::Zitadel;
+//
+// #[tokio::main]
+// async fn main() {
+//     let zitadel = Zitadel::default().start().await?;
+//     let host_port = zitadel_node.get_host_port_ipv4(zitadel::ZITADEL_PORT).await?;
+//     println!("Zitadel is running on port: {}", host_port);
+// }
+// ```
+
+/// Configuration for the Zitadel container.
+#[derive(Debug, Clone, Default)]
+pub struct Zitadel {
+    env_vars: HashMap<String, String>,
+}
+
+impl Zitadel {
+    // https://zitadel.com/docs/self-hosting/manage/configure
+
+    /// Configures external secure for the [`Zitadel`] instance.
+    /// ExternalSecure specifies if ZITADEL is exposed externally using HTTPS or HTTP.
+    /// Read more about external access: https://zitadel.com/docs/self-hosting/manage/custom-domain
+    /// Default is `true` if not overridden by this function
+    ///
+    /// See the [official docs for this option](https://zitadel.com/docs/self-hosting/manage/configure)
+    pub fn with_external_secure(mut self, external_secure: bool) -> Self {
+        if !external_secure {
+            self.env_vars
+                .insert("ZITADEL_EXTERNALSECURE".to_owned(), "false".to_owned());
+        }
```

Comment:
Once it's inserted, any additional call to `with_external_secure` won't have any effect I believe (sometimes it's convenient to override some defaults configured by external call for exaple)